packages/liveblocks-react/src/liveblocks.tsx

```diff
@@ -1179,6 +1199,127 @@ const _useUserThreads_experimental: TypedBundle["useUserThreads_experimental"] =
 const _useUserThreadsSuspense_experimental: TypedBundle["suspense"]["useUserThreads_experimental"] =
   useUserThreadsSuspense_experimental;
 
+function useSyncStatus_withClient(
+  client: OpaqueClient,
+  options?: UseSyncStatusOptions
+): SyncStatus {
+  // Normally the Rules of Hooks™ dictate that you should not call hooks
+  // conditionally. In this case, we're good here, because the same code path
+  // will always be taken on every subsequent render here, because we've frozen
+  // the value.
+  /* eslint-disable react-hooks/rules-of-hooks */
+  const smooth = useInitial(options?.smooth ?? false);
```

Comment:
Is there a particular reason why we freeze the value to always be the one set the first?

Reply:
Yes, if we did not do this, then calling `useSyncStatus({ smooth: true })` on the first render, and `useSyncStatus({ smooth: false })` on the second render would call a different hook, which would be problematic in React due to the fact that hooks rely on call order. (This is exactly the problem that the Rules of Hooks lint rule tries to prevent.) However, if we let the first render decide which branch is going to be taken for all subsequent renders, then this is not an issue.

Reply:
I was inquiring more from a design perspective. I understand that the way we have currently set up the hook, we want to absolutely freeze the `smooth` option but is it possible to set up the hook in a way that doesn't force `smooth` to be stabilized - maybe [if the `useSyncStatusSmooth_withClient` hook also uses `useSyncExternalStore`](https://github.com/liveblocks/liveblocks/pull/2043/files#r1836749015)?

Reply:
Did you mean to change the `getter`/`subscriber` as a pair, i.e. observe a _different_ external store based on the conditional?

```tsx
const subscribe = smooth ? smoothSubscribe : immediateSubscribe;
const getter = smooth ? smoothGetter : immediateGetter;
const status = useSyncExternalStore(subscribe, getter, getter);
```

I think that could work indeed! Would actually be quite nice 🤔 Maybe this is a good argument for creating that "deferred" external store anyway, like I [suggested here](https://github.com/liveblocks/liveblocks/pull/2043#discussion_r1837743310).

Anyway, I'll definitely keep this out of scope for this PR, but I agree it may be nice to set it up like this and be a better React citizen and avoid possible tearing use cases we're not aware of.